### PR TITLE
Add Form Action Tag Helper docs

### PR DESCRIPTION
**Pull request by NTaylorMullen**
- Utilized the `AnchorTagHelper` docs to consistenly describe the `asp-` attributes used to control how links get generated.
- Updated the built in TagHelper list doc to include the FormAction TagHelper.

#10593 
[Review URL](https://review.docs.microsoft.com/en-us/aspnet/core/mvc/views/working-with-forms?view=aspnetcore-2.2&branch=pr-en-us-11192#the-form-action-tag-helper)

### CODE REVIEW

File: aspnetcore/mvc/views/working-with-forms.md [hunk 1]
```diff
@@ -564,6 +564,95 @@ The correct `<option>` element will be selected ( contain the `selected="selecte
...
+
+|Attribute|Description|
+|---|---|
+|[asp-controller](xref:mvc/views/tag-helpers/builtin-th/anchor-tag-helper#asp-controller)|The name of the controller|
+|[asp-action](xref:mvc/views/tag-helpers/builtin-th/anchor-tag-helper#asp-action)|The name of the action method|
+|[asp-area](xref:mvc/views/tag-helpers/builtin-th/anchor-tag-helper#asp-area)|The name of the area|
+|[asp-page](xref:mvc/views/tag-helpers/builtin-th/anchor-tag-helper#asp-page)|The name of the Razor page|
+|[asp-page-handler](xref:mvc/views/tag-helpers/builtin-th/anchor-tag-helper#asp-page-handler)|The Razor page handler|
+|[asp-route](xref:mvc/views/tag-helpers/builtin-th/anchor-tag-helper#asp-route)|The name of the route|
+|[asp-route-{value}](xref:mvc/views/tag-helpers/builtin-th/anchor-tag-helper#asp-route-value)|A single URL route value i.e. `asp-route-id="1234"`|
```
Comment (scottaddie) [inline aspnetcore/mvc/views/working-with-forms.md]:
```suggestion
|[asp-route-{value}](xref:mvc/views/tag-helpers/builtin-th/anchor-tag-helper#asp-route-value)|A single URL route value. For example, `asp-route-id="1234"`.|
```

File: aspnetcore/mvc/views/working-with-forms.md [hunk 2]
```diff
@@ -564,6 +564,95 @@ The correct `<option>` element will be selected ( contain the `selected="selecte
...
+|[asp-all-route-data](xref:mvc/views/tag-helpers/builtin-th/anchor-tag-helper#asp-all-route-data)|All route values|
+|[asp-fragment](xref:mvc/views/tag-helpers/builtin-th/anchor-tag-helper#asp-fragment)|The URL fragment|
+
+### Submit to controller example
+
+The following example would submit the form to the `HomeController`s `Index` action if either the input or button were clicked.
```
Comment (scottaddie) [inline aspnetcore/mvc/views/working-with-forms.md]:
```suggestion
The following example submits the form to the `Index` action of `HomeController` if either the input or button were clicked:
```

File: aspnetcore/mvc/views/working-with-forms.md [hunk 3]
```diff
@@ -564,6 +564,95 @@ The correct `<option>` element will be selected ( contain the `selected="selecte
...
+
+```cshtml
+<form method="post">
+    <button asp-controller="Home" asp-action="Index">Click Me</button>
+    <input type="image" src="..." alt="Or Click Me" asp-controller="Home" asp-action="Index" />
+</form>
+```
+
+HTML Generation:
```
Comment (scottaddie) [inline aspnetcore/mvc/views/working-with-forms.md]:
```suggestion
The following HTML is generated:
```

File: aspnetcore/mvc/views/working-with-forms.md [hunk 4]
```diff
@@ -564,6 +564,95 @@ The correct `<option>` element will be selected ( contain the `selected="selecte
...
+The following example would submit the form to the `About` Razor Page.
+
+```cshtml
+<form method="post">
+    <button asp-page="About">Click Me</button>
+    <input type="image" src="..." alt="Or Click Me" asp-page="About" />
+</form>
+```
+
+HTML Generation:
```
Comment (scottaddie) [inline aspnetcore/mvc/views/working-with-forms.md]:
```suggestion
The following HTML is generated:
```

File: aspnetcore/mvc/views/working-with-forms.md [hunk 5]
```diff
@@ -564,6 +564,95 @@ The correct `<option>` element will be selected ( contain the `selected="selecte
...
+    <button formaction="/About">Click Me</button>
+    <input type="image" src="..." alt="Or Click Me" formaction="/About" />
+</form>
+```
+
+### Submit to route example
+
+The following example would submit the form to the `/Home/Test` endpoint.
+
+```C#
```
Comment (scottaddie) [inline aspnetcore/mvc/views/working-with-forms.md]:
```suggestion
```csharp
```

File: aspnetcore/mvc/views/working-with-forms.md [hunk 6]
```diff
@@ -564,6 +564,95 @@ The correct `<option>` element will be selected ( contain the `selected="selecte
...
+```
+
+```cshtml
+<form method="post">
+    <button asp-route="Custom">Click Me</button>
+    <input type="image" src="..." alt="Or Click Me" asp-route="Custom" />
+</form>
+```
+
+HTML Generation:
```
Comment (scottaddie) [inline aspnetcore/mvc/views/working-with-forms.md]:
```suggestion
The following HTML is generated:
```

File: aspnetcore/mvc/views/working-with-forms.md [hunk 7]
```diff
@@ -564,6 +564,95 @@ The correct `<option>` element will be selected ( contain the `selected="selecte
...
+```html
+<form method="post">
+    <button formaction="/Home">Click Me</button>
+    <input type="image" src="..." alt="Or Click Me" formaction="/Home" />
+</form>
+```
+
+### Submit to page example
+
+The following example would submit the form to the `About` Razor Page.
```
Comment (scottaddie) [inline aspnetcore/mvc/views/working-with-forms.md]:
```suggestion
The following example submits the form to the `About` Razor Page:
```

**Comment by scottaddie** [comment]
@NTaylorMullen Please also update the `ms.date` metadata value (at the top of the doc) to today's date.

**Comment by NTaylorMullen** [comment]
Thanks @scottaddie, 🆙 📅

**Comment by Rick-Anderson** [comment]
DO NOT MERGE until #11197 is merged into this PR.

File: aspnetcore/mvc/views/working-with-forms.md [hunk 8]
```diff
@@ -4,7 +4,7 @@ author: rick-anderson
 description: Describes the built-in Tag Helpers used with Forms.
 ms.author: riande
 ms.custom: mvc
-ms.date: 1/11/2019
+ms.date: 2/27/2019
 uid: mvc/views/working-with-forms
 ---
 # Tag Helpers in forms in ASP.NET Core
```
Comment (dougbu) [inline aspnetcore/mvc/views/working-with-forms.md]: Suggest adding "Nicholas T. Mullen" to the author list just below.

File: aspnetcore/mvc/views/working-with-forms.md [hunk 9]
```diff
@@ -564,6 +564,95 @@ The correct `<option>` element will be selected ( contain the `selected="selecte
...
+
+Supported [AnchorTagHelper](xref:mvc/views/tag-helpers/builtin-th/anchor-tag-helper) attributes to control the value of `formaction`:
+
+|Attribute|Description|
+|---|---|
+|[asp-controller](xref:mvc/views/tag-helpers/builtin-th/anchor-tag-helper#asp-controller)|The name of the controller.|
+|[asp-action](xref:mvc/views/tag-helpers/builtin-th/anchor-tag-helper#asp-action)|The name of the action method.|
+|[asp-area](xref:mvc/views/tag-helpers/builtin-th/anchor-tag-helper#asp-area)|The name of the area.|
+|[asp-page](xref:mvc/views/tag-helpers/builtin-th/anchor-tag-helper#asp-page)|The name of the Razor page.|
+|[asp-page-handler](xref:mvc/views/tag-helpers/builtin-th/anchor-tag-helper#asp-page-handler)|The Razor page handler.|
```
Comment (dougbu) [inline aspnetcore/mvc/views/working-with-forms.md]: Suggest `The name of the Razor page handler.` for consistency

**Comment by Rick-Anderson** [comment]
@NTaylorMullen 	Did you notice I copy/pasted from Form and added:

```
The Form Action Tag Helper:

* Generates the HTML [\<FORM>](https://www.w3.org/TR/html401/interact/forms.html) `action` attribute value for a MVC controller action, named route, or Razor Page.

* Generates a hidden [Request Verification Token](/aspnet/mvc/overview/security/xsrfcsrf-prevention-in-aspnet-mvc-and-web-pages) to prevent cross-site request forgery (when used with the `[ValidateAntiForgeryToken]` attribute in the HTTP Post action method)
``

**Comment by Rick-Anderson** [comment]
@NTaylorMullen 	let me know when it's ready to S&M

**Comment by NTaylorMullen** [comment]
> Did you notice I copy/pasted from Form and added:

Did not, the form action TagHelper does not generate a form or generate a request verification token. Both those statements should be removed.

**Comment by NTaylorMullen** [comment]
FormActionTagHelper truly just generates the `formaction` attribute on the generated `<button ...>` or `<input type="image" ...>` tag.

**Comment by Rick-Anderson** [comment]
@NTaylorMullen 	removed. Should we say something like that to differentiate it from the Form TH?

**Comment by Rick-Anderson** [comment]
`The Form Action Tag Helper controls where a form submits its data.`
Is probably good enough.

**Comment by Rick-Anderson** [comment]
Something like this?

The Form Action Tag Helper generates the `formaction` attribute on the generated `<button ...>` or `<input type="image" ...>` tag. The `formaction` attribute controls where a form submits its data.

**Comment by NTaylorMullen** [comment]
> Something like this?
> 
> The Form Action Tag Helper generates the `formaction` attribute on the generated `<button ...>` or `<input type="image" ...>` tag. The `formaction` attribute controls where a form submits its data.

Ya that sounds great!

**Comment by Rick-Anderson** [comment]
@NTaylorMullen 	ready for S&M?

**Comment by NTaylorMullen** [comment]
> @NTaylorMullen ready for S&M?

Go for it!

**Pull request merged**